lib/helpers.js

```diff
@@ -57,13 +57,24 @@ async function downloadApp (app, appExt) {
 }
 
 async function downloadFile (fileUrl, suffix) {
+
   // We will be downloading the files to a directory, so make sure it's there
   // This step is not required if you have manually created the directory
   let fileInfo = await tempDir.open({prefix: 'appium-app', suffix: suffix});
   _fs.close(fileInfo.fd);
-  await fs.writeFile(fileInfo.path, await request(fileUrl));
-  logger.debug(`${fileUrl} downloaded to ${fileInfo.path}`);
 
+  // we use the non-promise version of request because the 'pipe' function
+  // won't overuse memory. request-promise has to store the whole response
+  // in memory (and also has encoding trickiness)
+  let writeStream = _fs.createWriteStream(fileInfo.path);
+  writeStream.on('error', logger.errorAndThrow);
```

Comment:
I am not sure this will throw in the correct context. Have you tested it? My guess is it will throw an unhandled error. Probably we need to re-think this and wrap the whole thing in a Promise, and reject if we hit an error event.